Guided tutorial

StevieGoodman opened the pull request.

HFinch13 committed:
The Slope is a slope now

the slope is now actually a slope. and along with this the doors work. they are deactivated by something with tag "Player" hitting the connected button

HFinch13 committed:
Merge branch 'guided_tutorial' of https://github.com/CMP2804/Assessment3 into guided_tutorial

kit-calloway commented on Assets/ButtonBehaviour.cs:
```diff
@@ -0,0 +1,36 @@
+    public class ButtonBehaviour : MonoBehaviour
+    {
+
+        public GameObject Player;
```
This can be removed

```suggestion
```

HFinch13 committed:
Geometry fix

Fixed some geometry, and removed the door testing things so that it doesn't activate on its own now

kit-calloway commented on Assets/ButtonBehaviour.cs:
```diff
@@ -0,0 +1,36 @@
+
+        public GameObject Player;
+        public GameObject Button;
+        
+        // Start is called before the first frame update
+        void Start()
+        {
+
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            
+        }
```
All of this can be removed.
```suggestion
```

kit-calloway commented on Assets/ButtonBehaviour.cs:
```diff
@@ -0,0 +1,36 @@
+            {
+            
+                Button.tag = "pressed";
+            }
```
This can be simplified:
```suggestion
            if (coll.gameObject.tag == "Player")
                Button.tag = "pressed";
```

kit-calloway commented on Assets/ButtonLight.cs:
```diff
@@ -0,0 +1,30 @@
+        // Update is called once per frame
+        void Update()
+        {
+            if (button.tag == "pressed")
+            {
+                UnityEngine.Debug.Log("BUTTON PRESSED");
+                button.GetComponent<Renderer>().material = green;
+            }
+            
```
Indentation can be removed by inverting conditional.
```suggestion
            if (button.tag == "pressed") return;
            button.GetComponent<Renderer>.material = green;
```

kit-calloway commented on Assets/ButtonLight.cs:
```diff
@@ -0,0 +1,30 @@
+
+        public Material green;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+        
+        }
+
```
This can be removed

```suggestion
```

kit-calloway commented on Assets/DoorBehaviour.cs:
```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+
+
+
+
```
Remove empty lines
```suggestion
```

kit-calloway commented on Assets/DoorBehaviour.cs:
```diff
@@ -0,0 +1,53 @@
+        public GameObject Button;
+        public GameObject Button2;
+
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            
+        }
+
```
This can be removed
```suggestion
```

kit-calloway commented on Assets/DoorBehaviour.cs:
```diff
@@ -0,0 +1,53 @@
+        // Update is called once per frame
+        void Update()
+        {
+            if (Door.gameObject.tag == "Door")
+            {
+                if (Button.tag == "pressed")
+                {
+                    Door.SetActive(false);
+                }
+            }
+            
+            else if (Door.tag == "Doorv2")
+            {
+                
+               
+                if (Button.tag == "pressed" && Button2.tag == "pressed" )
+                {
+                    Door.SetActive(false);
+                }
+                
+                
+            }
+           
+
```
This can be improved.

```suggestion
            if (Door.gameObject.tag == "Door" && Button.tag == "pressed")
                Door.SetActive(false);
            else if (Door.tag == "Doorv2" && Button.tag == "pressed" && Button2.tag == "pressed")
                Door.SetActive(false);
```

StevieGoodman committed:
Applied suggestions from code review

Signed-off-by: Stevie Goodman <mail@steviegoodman.com>

StevieGoodman committed:
Fix branch

Signed-off-by: Stevie Goodman <mail@steviegoodman.com>

HFinch13 committed:
Stairs Begone

Got Rid of them stairs/slopes

HFinch13 committed:
Merge branch 'guided_tutorial' of https://github.com/CMP2804/Assessment3 into guided_tutorial

StevieGoodman reviewed (commented):
Please clean up your project hierarchy. You have 4 scenes here, 2 prefabs and none of them are the guided tutorial level with no stairs. Please name the scene "Tutorial Level" and place it in `Assets/Scenes`.
Please also place scripts in `Assets/Scripts/Buttons`.

HFinch13 committed:
Geometry changes

geometry changes in line with making it better

StevieGoodman committed:
Merge branch 'submission' into guided_tutorial

Signed-off-by: Stevie Goodman <mail@steviegoodman.com>
